src/Hosting/Hosting/src/Internal/HostingRequestStartingLog.cs

```diff
@@ -51,25 +60,24 @@ internal class HostingRequestStartingLog : IReadOnlyList<KeyValuePair<string, ob
 
         public HostingRequestStartingLog(HttpContext httpContext)
         {
-            _request = httpContext.Request;
+            var request = httpContext.Request;
+
+            Protocol = request.Protocol;
+            Method = request.Method;
+            ContentType = !string.IsNullOrEmpty(request.ContentType) ? Uri.EscapeUriString(request.ContentType) : string.Empty;
+            ContentLength = request.ContentLength;
+            Scheme = request.Scheme;
+            Host = request.Host.Value;
+            PathBase = request.PathBase.Value;
+            Path = request.Path.Value;
+            QueryString = request.QueryString.Value;
         }
 
         public override string ToString()
         {
             if (_cachedToString == null)
             {
-                _cachedToString = string.Format(
-                    CultureInfo.InvariantCulture,
-                    "Request starting {0} {1} {2}://{3}{4}{5}{6} {7} {8}",
-                    _request.Protocol,
-                    _request.Method,
-                    _request.Scheme,
-                    _request.Host.Value,
-                    _request.PathBase.Value,
-                    _request.Path.Value,
-                    _request.QueryString.Value,
-                    _request.ContentType,
-                    _request.ContentLength);
+                _cachedToString = $"{LogPreamble}{Protocol} {Method} {Scheme}://{Host}{PathBase}{Path}{QueryString} {ContentType} {ContentLength?.ToString(CultureInfo.InvariantCulture) ?? string.Empty}";
```

Comment:
```suggestion
                _cachedToString = $"{LogPreamble}{Protocol} {Method} {Scheme}://{Host}{PathBase}{Path}{QueryString} {ContentType} {ContentLength?.ToString(CultureInfo.InvariantCulture) ?? string.Empty}".TrimEnd(' ');
```

Reply:
Would change the column positions of the fields based on which ones are filled in?